kong/dao/schemas/plugins.lua

```diff
@@ -1,14 +1,31 @@
 local utils = require "kong.tools.utils"
 local Errors = require "kong.dao.errors"
 local db_errors = require "kong.db.errors"
+local singletons = require "kong.singletons"
+local constants = require "kong.constants"
 
 local function load_config_schema(plugin_t)
-  if plugin_t.name then
-    local loaded, plugin_schema = utils.load_module_if_exists("kong.plugins." .. plugin_t.name .. ".schema")
+  local plugin_name = plugin_t.name
+
+  if plugin_name then
+
+    if constants.DEPRECATED_PLUGINS[plugin_name] then
+      ngx.log(ngx.WARN, "the '" .. plugin_name .. "' plugin has been deprecated")
+    end
+
+    -- singletons.configuration would be nil when plugin operations are
+    -- done through DAOs like in migrations or tests
+    if singletons.configuration and not singletons.configuration.plugins[plugin_name] then
+      return nil, 'Plugin "' .. tostring(plugin_name) ..
+      '" not enabled, consider adding it with custom_plugins configuration'
```

Comment:
we do not capitalize error messages (ever) - especially since they will generally be printed after a colon `something went wrong: <error msg>`. Also better to stay consistent with regards to single/double quotes usage.

Besides, the indentation need to be reworked too, perhaps:
```lua
return nil "plugin '" .. plugin_name .. "' not enabled; " ..
           "add it to the 'custom_plugins' configuration property"
```